contracts/src/systems/combat/contracts.cairo

```diff
@@ -686,6 +687,27 @@ mod combat_systems {
 
             set!(world, (battle));
 
+            let id = world.uuid();
+
+            let attacker = starknet::get_caller_address();
+            let defender = get!(world, defending_army_id, Owner).address;
+            emit!(
+                world,
+                EternumEvent {
+                    id,
+                    event_id: EventType::BattleStart,
+                    timestamp: starknet::get_block_timestamp(),
+                    data: EventData::BattleStart(
+                        BattleStartData {
+                            attacker,
+                            defender,
+                            attacker_army_entity_id: attacking_army_id,
+                            defender_army_entity_id: defending_army_id,
+                            duration_left: 100,
```

Comment:
Do we even need this attribute in battleStartData ?

Reply:
We could compute it in the client or here. It might make sense to do it here so we don't duplicate and have another dep.